src/main/java/com/epam/controller/TraineeController.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.controller;
+
+import com.epam.dto.trainee.*;
+import com.epam.dto.trainer.TrainerProfile;
+import com.epam.mappers.TraineeMapper;
+import com.epam.model.Trainee;
+import com.epam.model.Trainer;
+import com.epam.model.Training;
+import com.epam.service.TraineeService;
+import com.epam.service.TrainerService;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/trainee")
+@RequiredArgsConstructor
+public class TraineeController {
+    private final TraineeMapper mapper;
+    private final TraineeService traineeService;
+    private final TrainerService trainerService;
+
+    @PostMapping
+    public ResponseEntity<TraineeRegistrationResponse> registerTrainee(@RequestBody @Valid TraineeRegistrationRequest request) {
+        Trainee trainee = mapper.traineeRegistrationRequestToTrainee(request);
+        Trainee createdTrainee = traineeService.createTrainee(trainee);
+        TraineeRegistrationResponse response = new TraineeRegistrationResponse(
+                createdTrainee.getUser().getUsername(),
+                createdTrainee.getUser().getPassword());
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping
+    @Transactional
```

Comment:
Dear Garik! Please avoid making controller methods `@Transactional` - read this please - https://itecnote.com/tecnote/java-why-we-shouldnt-make-a-spring-mvc-controller-transactional/